### PR TITLE
add skip-blank-lines option to no-trailing-whitespace

### PR DESCRIPTION
**Pull request by cyrilgandon**
#### PR checklist

- [x] Addresses an existing issue: #481
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [ ] Documentation update

#### Is there anything you'd like reviewers to focus on?

Should the option be called `skip-blank-lines` like [eslint](https://eslint.org/docs/rules/no-trailing-spaces), or `ignore-blank-lines`, like every other option of the rule?

#### CHANGELOG.md entry:

[new-rule-option] [`no-trailing-whitespace`](https://palantir.github.io/tslint/rules/no-trailing-whitespace/) : `ignore-blank-lines`

### CODE REVIEW

File: src/rules/noTrailingWhitespaceRule.ts [hunk 1]
```diff
@@ -24,11 +24,13 @@ import { getTemplateRanges } from "./noConsecutiveBlankLinesRule";
 const OPTION_IGNORE_COMMENTS = "ignore-comments";
 const OPTION_IGNORE_JSDOC = "ignore-jsdoc";
 const OPTION_IGNORE_TEMPLATE_STRINGS = "ignore-template-strings";
+const OPTION_SKIP_BLANK_LINES = "skip-blank-lines";
```
Comment (ajafff) [inline src/rules/noTrailingWhitespaceRule.ts]: Please rename to `"ignore-blank-lines"` for consistency

File: src/rules/noTrailingWhitespaceRule.ts [hunk 2]
```diff
@@ -78,12 +82,15 @@ function walk(ctx: Lint.WalkContext<Options>) {
     const sourceFile = ctx.sourceFile;
     const text = sourceFile.text;
     for (const line of getLineRanges(sourceFile)) {
-        const match = text.substr(line.pos, line.contentLength).match(/\s+$/);
-        if (match !== null) {
```
Comment (ajafff) [inline src/rules/noTrailingWhitespaceRule.ts]: You don't need a separate function or regex.
`match[0]` contains the full match. If the length of the match is equal to the line length, the line consists only of whitespace.
```ts
if (match !== null && (!ctx.options.skipBlankLines || match[0].length !== line.contentLength))
```

Comment (ajafff) [reply src/rules/noTrailingWhitespaceRule.ts]: or even simpler: check if `match.index === 0`

**Comment by ajafff** [comment]
CI failure is unrelated because of upstream changes in typescript@next
Will try to fix this in a separate PR. Until then I cannot merge this PR.

**Comment by ajafff** [comment]
Thanks @cyrilgandon

**Comment by gonzofish** [comment]
Awesome!

**Comment by cmartin88** [comment]
Has this been implemented? Not seeing it in webstorm.

**Pull request merged**